components/generate_embeddings/src/main.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+import pandas as pd
+from fondant.component import PandasTransformComponent
+from langchain.embeddings import (
+    AlephAlphaAsymmetricSemanticEmbedding,
+    CohereEmbeddings,
+    HuggingFaceEmbeddings,
+    OpenAIEmbeddings,
+)
+from retry import retry
+from utils import secrets_to_env_vars
+
+logger = logging.getLogger(__name__)
+
+
+class GenerateEmbeddings(PandasTransformComponent):
+    def __init__(
+        self,
+        *_,
+        model: str,
+        model_provider: str,
+    ):
+        self.model = model
+        self.model_provider = model_provider
+
+        # to pass embedding model API keys via env vars
+        secrets_to_env_vars()
```

Comment:
For now I think the easiest way would be to pass the secret as component argument to the component. 
The goal of the reusable components is that you can use it out of the box. This wouldn't work here, if a user has to adapt the code inside of the component.